src/mixins/itext_click_behavior.mixin.ts

```diff
@@ -192,6 +195,8 @@ export abstract class ITextClickBehaviorMixin<
       return;
     }
 
+    shouldSetCursor && this.setCursorByClick(options.e);
```

Comment:
What is unclear to me is why this shouldn't have triggered on mouse down.
Not a big deal, this is not relevant to merging.
Can you try to explain anyway this is here? Setting cursor was usually mouse down tas

Reply:
I see the mouseDown has an extra `this.__isDragging ||` how do we trigger dragging before mouse down event?

Reply:
`_mouseDownHandlerBefore` sets `this.__isDragging` to keep selection, it is clearer in #8598 https://github.com/fabricjs/fabric.js/blob/27dcd9e011f7f4fd235e8a7490c2e1775f68fd40/src/mixins/DraggableTextDelegate.ts#L12-L21 )

Reply:
ok. so mosedown:before shoud allow for the developer to interact before fabric does anything.
That was the point of that event.
Let's try to not break that contract.

Reply:
So you suggest moving logic to mouse down?
Sounds correct

Reply:
2aadf5d53